### PR TITLE
Backport CVE-2021-34551 and CVE-2021-3603 for PHPMailer

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue https://github.com/joomla-backports/PHPMailer/pull/5

### Summary of Changes

Backported the fixes for CVE-2021-34551 and CVE-2021-3603

### Testing Instructions
Make sure the phpmailer still works

### Actual result BEFORE applying this Pull Request

CVE-2020-34551 and CVE-2021-3603

### Expected result AFTER applying this Pull Request

CVE-2020-34551 and CVE-2021-3603 patched

### Documentation Changes Required

none

### CODE REVIEW

**Comment by richard67** [comment]
@zero-24 Unit tests are failing for PHP 5.3, see e.g. https://ci.joomla.org/joomla/joomla-cms/45817/2/8 :

```
PHP Parse error:  syntax error, unexpected '[' in /drone/src/libraries/vendor/phpmailer/phpmailer/class.phpmailer.php on line 1826
```

Can it be that the backport uses short array syntax which is not supported in these old PHP versions?

File: libraries/vendor/phpmailer/phpmailer/class.phpmailer.php [hunk 1]
```diff
@@ -1815,9 +1816,27 @@ public function setLanguage($langcode = 'en', $lang_path = '')
...
-                // This way we'll never have missing translation keys.
-                $foundlang = include $lang_file;
+                //$foundlang = include $lang_file;
+                $lines = file($lang_file);
+                foreach ($lines as $line) {
+                    //Translation file lines look like this:
+                    //$PHPMAILER_LANG['authenticate'] = 'SMTP-Fehler: Authentifizierung fehlgeschlagen.';
+                    //These files are parsed as text and not PHP so as to avoid the possibility of code injection
+                    //See https://blog.stevenlevithan.com/archives/match-quoted-string
+                    $matches = [];
```
Comment (zero-24) [inline libraries/vendor/phpmailer/phpmailer/class.phpmailer.php]:
```suggestion
                    $matches = array();
```

**Comment by zero-24** [comment]
That should fix it if so i'm doing an upstream PR and update here too.

**Comment by zero-24** [comment]
Drone is happy now will update it on the repo and update here: https://ci.joomla.org/joomla/joomla-cms/45819/2/8

**Comment by zero-24** [comment]
Branch updated with the latest changes from the backports repo.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 1a78cd5bac9b0cf395dec2b377b9ad86d1255cf2<br /><br />I've successfully tested this PR with both methods "PHP Mail" and "SMTP".<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34783">issues.joomla.org/tracker/joomla-cms/34783</a>.</sub>

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 1a78cd5bac9b0cf395dec2b377b9ad86d1255cf2<br /><br />tested with method "PHP Mail" <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34783">issues.joomla.org/tracker/joomla-cms/34783</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34783">issues.joomla.org/tracker/joomla-cms/34783</a>.</sub>

**Comment by zero-24** [comment]
Merging for now thanks!

**Pull request merged**